cli/tsc.rs

```diff
@@ -525,8 +531,69 @@ impl TsCompiler {
       return Err(ErrBox::from(compile_response.diagnostics));
     }
 
+    self.set_graph_metadata(
+      source_file.url.clone(),
+      &compile_response.emit_map,
+    )?;
     self.cache_emitted_files(compile_response.emit_map)?;
-    ts_compiler.get_compiled_module(&source_file_.url)
+    Ok(())
+  }
+
+  fn get_graph_metadata(&self, url: &Url) -> Option<GraphFileMetadata> {
+    // Try to load cached version:
+    // 1. check if there's 'meta' file
+    let cache_key = self
+      .disk_cache
+      .get_cache_filename_with_extension(url, "graph");
+    if let Ok(metadata_bytes) = self.disk_cache.get(&cache_key) {
+      if let Ok(metadata) = std::str::from_utf8(&metadata_bytes) {
+        if let Ok(read_metadata) =
+          GraphFileMetadata::from_json_string(metadata.to_string())
+        {
+          return Some(read_metadata);
+        }
+      }
+    }
+
+    None
+  }
+
+  fn set_graph_metadata(
+    &self,
+    url: Url,
+    emit_map: &HashMap<String, EmittedSource>,
+  ) -> std::io::Result<()> {
+    let version_hash =
+      crate::checksum::gen(vec![version::DENO.as_bytes(), &self.config.hash]);
+    let mut deps = vec![];
```

Comment:
This function should discriminate files that were compiled to JS as well as other auxiliary files (like type definitions)